katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,43 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  return unless facts[:os][:release][:major] == '7'
+
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  current_postgres_dir = '/var/lib/pgsql'
+  new_postgres_dir = '/var/opt/rh/rh-postgresql10/lib/pgsql'
+
+  begin
+    unless File.exist?(new_postgres_dir)
```

Comment:
Any reason you're not using my code? My version detected the separate partition `/var` for example where yours doesn't.

Reply:
I was not clear from looking at your code `how` it detected the separate partition without erroring when it did not exist _and_ then checked the base filesystem.

Reply:
It started with the directory and went up until the directory actually exists. You can give `df` any directory and it'll find the correct mountpoint as long as it exists. If you keep going up, you'll end up with `/` and that always exists:
```
[1] pry(main)> File.dirname('/doesnotexist')
=> "/"
```

Reply:
Not on my systems, which is part of why I didn't go that route:

```
irb(main):001:0> File.dirname('/var/foo/var/this/I/am/not/real')
=> "/var/foo/var/this/I/am/not"
```

Reply:
That's why there was the `until` loop :)
```
[1] pry(main)> directory = '/var/foo/var/this/I/am/not/real'
=> "/var/foo/var/this/I/am/not/real"
[2] pry(main)> until File.directory?(directory)
[2] pry(main)*   directory = File.dirname(directory)
[2] pry(main)* end  
=> nil
[3] pry(main)> puts directory
/var
=> nil
```